owa/app/js/components/manage_patient_identifier_sources/existingIdentifierSources.jsx

```diff
@@ -0,0 +1,95 @@
+import React, {Component} from 'react';
+import {Table, Button, Modal, ModalHeader, ModalBody, ModalFooter} from 'reactstrap';
+import ViewSource from './viewSource';
+import ConfigureSource from './configureSource';
+
+export default class ExistingIdentifierSources extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+      modalContent: '',
+      identifierSources: [{
+        identifierType: ['OpenMRS ID'],
+        sourceType: ['Local Identifier Generator'],
+        sourceName: ['New Source']
+      }]
+    };
+
+    this.toggle = this.toggle.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  //show the modal
+  toggle() {
+    this.setState({
+      modal: !this.state.modal
+    });
+  }
+
+  //handle clicks on the buttons
+  handleClick(event){
+    if(event.target.name === 'configureButton'){
+      this.setState({
+        modalContent: <ConfigureSource />
```

Comment:
Why are you setting the state to a component? Am I missing something?

Reply:
The content of the modal changes depending on which button is clicked (Configure or View). I am using the state to put the appropriate content depending on the button clicked. Your thoughts?